qemu/tests/image_create_with_preallocation.py

```diff
@@ -0,0 +1,78 @@
+import json
+import os
+
+from avocado import fail_on
+from avocado.utils import process
+
+from provider import qemu_img_utils as img_utils
+from virttest import data_dir
+from virttest import qemu_storage
+
+
+def run(test, params, env):
+    """
+    Check parameter preallocation when creating an image.
+    1. Create a qcow2 image with preallocation=off, full, falloc, metadata.
+    2. Create a raw image with preallocation=off, full, falloc.
+    3. Create a luks image with preallocation=off, full, falloc.
+    4. check points:
+    4.1 preallocation=off
+    Image create successfully, the actual_size is less than specified size.
+    4.2 preallocation=full
+    Image create successfully, the actual_size is greater than or equal to
+    specified size.
+    (It's normal because of the temporary predictive preallocation of XFS.)
+    4.3 preallocation=falloc
+    Image create successfully, invoked fallocate system call,
+    the actual_size is greater than or equal to specified size.
+    (It's normal because of the temporary predictive preallocation of XFS.)
+    4.4 preallocation=metadata
+    Image create successfully, the actual_size is less than specified size.
+
+    :param test: VT test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def check_fallocate_syscall(trace_events):
+        """
+        check whether invoke fallocate system call
+        when creating an image with preallocation=falloc
+        """
+        strace_log = os.path.join(test.debugdir, "fallocate.log")
+        with img_utils.strace(img_stg, trace_events.split(), strace_log, True):
+            fail_on((process.CmdError,))(img_stg.create)(image_stg_params)
+        with open(strace_log) as fd:
+            if trace_events not in fd.read():
```

Comment:
if we only trace one event, it might be better to rename it to trace_event;
if we might extend to trace more than one, here we might need a loop;

Reply:
Only trace one event, I will rename it to trace_event. Thanks